code/parse/sexp.h

```diff
@@ -27,109 +27,121 @@ struct ship_obj;
+	OPF_SUBSYSTEM_OR_NONE,			// Goober5000 - an "optional" subsystem argument
+	OPF_PERSONA,					// Karajorma - name of a persona
+	OPF_SUBSYS_OR_GENERIC,			// Karajorma - a subsystem or a generic name (like engine) which covers all subsystems of that type
+	OPF_ORDER_RECIPIENT,			// Karajorma - since orders can go to All Fighters as well as a ship or wing
+	OPF_SUBSYSTEM_TYPE,				// Goober5000 - a generic subsystem type (navigation, engines, etc.) rather than a specific subsystem
+	OPF_POST_EFFECT,				// Hery - type of post-processing effect
+	OPF_TARGET_PRIORITIES,			// FUBAR - Target priority groups
+	OPF_ARMOR_TYPE,					// FUBAR - Armor type or <none>
+	OPF_FONT,						// Goober5000 - a FreeSpace font
+	OPF_HUD_ELEMENT,				// A magic name of a specific HUD element
+	OPF_SOUND_ENVIRONMENT,			// Goober5000 - one of EFX_presets, per Taylor
+	OPF_SOUND_ENVIRONMENT_OPTION,	// Goober5000 - one of Taylor's options
+	OPF_EXPLOSION_OPTION,			// Goober5000
+	OPF_AUDIO_VOLUME_OPTION,		// The E
+	OPF_WEAPON_BANK_NUMBER,			// Karajorma - The number of a primary/secondary/tertiary weapon bank or all of them
+	OPF_MESSAGE_OR_STRING,			// Goober5000 - provides a list of messages like OPF_MESSAGE, but also allows entering arbitrary strings
+	OPF_BUILTIN_HUD_GAUGE,			// The E
+	OPF_DAMAGE_TYPE,				// FUBAR - Damage type or <none>
+	OPF_SHIP_EFFECT,				// The E - per-ship effects, as defined in post-processing.tbl
+	OPF_ANIMATION_TYPE,				// Goober5000 - as defined in modelanim.h
+	OPF_MISSION_MOOD,				// Karajorma - Moods determine which builtin messages will be sent
+	OPF_SHIP_FLAG,					// Karajorma - The name of a ship flag
+	OPF_TEAM_COLOR,					// The E - Color settings as defined in Colors.tbl
+	OPF_NEBULA_PATTERN,				// Axem - Full Nebula Background Patterns, as defined in nebula.tbl
+	OPF_SKYBOX_FLAGS,				// niffiwan - valid skybox flags
+	OPF_GAME_SND,					// m!m - A game sound
+	OPF_FIREBALL,					// Goober5000 - an entry in fireball.tbl
+	OPF_SPECIES,					// Goober5000
+	OPF_LANGUAGE,					// Goober5000
+	OPF_FUNCTIONAL_WHEN_EVAL_TYPE,	// Goober5000
+	OPF_CONTAINER_NAME,				// Karajorma/jg18 - The name of a SEXP container
+	OPF_LIST_CONTAINER_NAME,		// Karajorma/jg18 - The name of a SEXP list container
+	OPF_MAP_CONTAINER_NAME,			// Karajorma/jg18 - The name of a SEXP map container
+	OPF_ANIMATION_NAME,				// Lafiel
+	OPF_CONTAINER_VALUE,			// jg18 - Container data and map container keys
+	OPF_DATA_OR_STR_CONTAINER,		// jg18 - any data, or a container that is accessed via strings
+	OPF_TRANSLATING_SUBSYSTEM,		// Goober5000 - a translating subsystem
+	OPF_ANY_HUD_GAUGE,				// Goober5000 - both custom and builtin
+	OPF_WING_FLAG,					// Goober5000 - The name of a wing flag
+	OPF_ASTEROID_DEBRIS,			// MjnMixael - Debris types as defined in asteroids.tbl
+	OPF_WING_FORMATION,				// Goober5000 - as defined in ships.tbl
+
+	//Must always be at the end of the list
+	First_available_list_id
```

Comment:
I would recommend calling this `First_available_enum_list_id` to match the other change.  Or perhaps `First_available_opf_id`.

Reply:
Enum is not strictly accurate and opf isn't really understandable to anyone not familiar with the code, but ok whatever.

Reply:
I'm open to other suggestions.  I like `First_available_opf_id` because opf_ is distinct enough to come up easily in a code search.